### PR TITLE
ZBUG-3064: X-Zimbra-Forwarded is not added by SieveFilter rule due to…

### PR DESCRIPTION
**Pull request by vishnukantjangid**
**Issue:** X-Zimbra-Forwarded is not added by SieveFilter rule due to ParseException

**Solution:** If we are getting mail ParseException during saving changes and X-Zimbra-Forwarded header is not present in the mail header then we are adding X-Zimbra-Forwarded in the mail to detect the mail loop.

### CODE REVIEW

**Review by zimsuchitgupta — APPROVED** [review]
LGTM

File: store/src/java/com/zimbra/cs/filter/FilterUtil.java [hunk 1]
```diff
@@ -290,13 +290,23 @@ public static void redirect(OperationContext octxt, Mailbox sourceMbox, MimeMess
             try {
                 outgoingMsg = createRedirectMsgOnError(msg);
                 ZimbraLog.filter.info("Message format error detected.  Wrapper class in use.  %s", e.toString());
+                // ZBUG-3064: If MimeMessage.saveChanges fails, updating header in copy of the message
```
Comment (shrutig0510) [inline store/src/java/com/zimbra/cs/filter/FilterUtil.java:293]: looks like redundant code. Create separate bock and use it at both places

Comment (vishnukantjangid) [reply store/src/java/com/zimbra/cs/filter/FilterUtil.java]: done

**Review by yyoshiho — COMMENTED** [review]
This fixes mail loop issue, but redirected message still has wrong headers because they are copied from original message. (This test case is wrong Content-Type with charset.) 
Should we follow to rfc2045? to avoid some mail problems with wrong headers for other messaging systems for associated addresses.

**Review by yyoshiho — APPROVED** [review]
This particular invalid Content-Type case is resolved by this fix.

**Pull request merged**